File: Data Structures & Algorithms/subsets/submission-2.cpp
class Solution {
public:
        vector<vector<int>> soln;

    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> temp;
        backTrack(nums,0,temp);
        return soln;        
    }

    void backTrack(vector<int>& nums,int index,vector<int>& temp){
        
        if(index==nums.size())
        soln.push_back({temp});

        if(index<nums.size()){
            temp.push_back(nums[index]);
            backTrack(nums,index+1,temp);
            temp.pop_back();
            backTrack(nums,index+1,temp);
        }

    }

};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int len=nums.length,l=0,r=len-1,m;

        while(l<r && r<len){
            m=(l+r)/2;

            if(nums[m]>nums[m+1])
                return nums[m+1];

            if(m!=0 && nums[m]<nums[m-1])
                return nums[m];

            if(nums[m]>nums[l])
                l=m+1;
            
            if(nums[m]<nums[r])
                r=m-1;

        }
        return nums[0];
    }
}
